### PR TITLE
Fix load 32 bits

### PR DESCRIPTION
**Pull request by guillep**
- Fix generation of Load 32 bits in ARMv8 backend with offsets != 0
- Add support for Armv7 and lower for Load 23 bits
- Add tests to ensure that values are zero extended by default (in 64 bits)

### CODE REVIEW

File: smalltalksrc/VMMakerTests/VMCogitHelpersTest.class.st [hunk 1]
```diff
@@ -127,3 +127,49 @@ VMCogitHelpersTest >> testCheckSmallIntegerWithValidSmallIntegers [
 	self assertIsInSmallIntegerRange: 0.
 	self assertIsInSmallIntegerRange: memory maxSmallInteger
 ]
+
+{ #category : 'as yet unclassified' }
```
Comment (guillep) [inline smalltalksrc/VMMakerTests/VMCogitHelpersTest.class.st]:
```suggestion
{ #category : 'testing' }
```

File: smalltalksrc/VMMakerTests/VMCogitHelpersTest.class.st [hunk 2]
```diff
@@ -127,3 +127,49 @@ VMCogitHelpersTest >> testCheckSmallIntegerWithValidSmallIntegers [
...
+	stub := self compile: [ 
+		cogit MoveM32: 8 r: ReceiverResultReg R: ReceiverResultReg.
+		cogit RetN: 0 ].
+	
+	self runUntilReturnFrom: stub.
+
+	self assert: (machineSimulator receiverRegisterValue) equals: 16rFFFFFFFF
+]
+
+{ #category : 'as yet unclassified' }
```
Comment (guillep) [inline smalltalksrc/VMMakerTests/VMCogitHelpersTest.class.st]:
```suggestion
{ #category : 'testing' }
```
